src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -195,9 +199,13 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
                 _packageRestoreManager.PackageRestoreFailedEvent -= PackageRestoreManager_PackageRestoreFailedEvent;
 
                 TelemetryUtility.StopTimer();
-
                 var duration = TelemetryUtility.GetTimerElapsedTime();
-                await _logger.WriteSummaryAsync(_status, duration);
+
+                // Do not log any restore message if user disabled restore.
+                if (_packageRestoreConsent.IsGranted)
+                {
+                    await _logger.WriteSummaryAsync(_status, duration);
```

Comment:
Do we really want to stop silently?

I think it's important to have some sort of message when we for negative conditions like this one. 
Something like a warning message, or just a log statement at the debug level to avoid being too noisy.